edb/lib/std/20-genericfuncs.edgeql

```diff
@@ -73,6 +73,27 @@ std::assert_distinct(
     USING SQL EXPRESSION;
 };
 
+# std::assert -- boolean assertion
+# --------------------------------
+CREATE FUNCTION
+std::assert(
+    input: bool,
+    NAMED ONLY message: OPTIONAL str = <str>{},
+) -> bool
+{
+    CREATE ANNOTATION std::description :=
+        "Assert that a boolean value is true.";
+    SET volatility := 'Stable';
+    USING SQL $$
+    SELECT (
+        edgedb.raise_on_null(
+            nullif("input", false),
+            'assert_failure',
```

Comment:
Hm.  Not sure it's totally safe to use this.  Postgres may throw this as well (though currently only seems to be produced by `assert` in pl/pgSQL.  To be completely safe, I would use the same technique we use for other assertions: use a constraint violation error and specify `std::assert` as the constraint (`raise` has a named arg for the constraint name).